### PR TITLE
Validate that `'self'` is present in the init signature of a Parameterized class

### PR DESCRIPTION
**Pull request by maximlt**
Fixes https://github.com/holoviz/param/issues/785

### CODE REVIEW

File: tests/testparameterizedrepr.py [hunk 1]
```diff
@@ -217,3 +218,38 @@ def test_script_repr_parameterized_instance(P):
...
+        # as P doesn't define __init__
+
+        # bad
+        P.__init__.__signature__ = inspect.Signature(
+            [
+                inspect.Parameter('test', inspect.Parameter.KEYWORD_ONLY),
+            ]
+        )
+
+        with pytest.raises(KeyError, match="'T.__init__.__signature__' must contain a 'self' Parameter."):
```
Comment (Hoxbro) [inline tests/testparameterizedrepr.py:241]: I would escape `.` with `\.`

**Review by Hoxbro — COMMENTED** [review]
I would maybe make a subclass with a custom `__init__` and use that as a base to not mess around with `param.Parameterized` directly.

Maybe also have an example with a parameter in it with a non-default (If it does not already exist.)

This still needs a change in Holoviews, correct?

**Comment by maximlt** [comment]
I addressed your comments.

> Maybe also have an example with a parameter in it with a non-default (If it does not already exist.)

`testparameterizedrepr` has a number of `.param.pprint` tests, including testing classes with an `__init__` that has positional only arguments.

File: tests/testparameterizedrepr.py [hunk 2]
```diff
@@ -217,3 +218,39 @@ def test_script_repr_parameterized_instance(P):
...
+
+    class P(param.Parameterized):
+        def __init__(self, **params): pass
+
+    class T(P): pass
+
+    t = T()
+
+    # This is actually setting the signature of param.Parameterized.__init__
+    # as P doesn't define __init__
```
Comment (Hoxbro) [inline tests/testparameterizedrepr.py:234]:
```suggestion
    # This is actually setting the signature of P.__init__
    # as T doesn't define __init__
```

**Review by Hoxbro — APPROVED** [review]

**Pull request merged**